src/components/NewItemRow/index.jsx

```diff
@@ -0,0 +1,116 @@
+
+    const [itemInfo, setItemInfo] = useState({
+        description: "",
+        quantity: 0,
+        rate: 0.00,
+        total: 0.00,
+    });
+    const rowTotal = (quantity, rate) => {
+        return quantity && rate
+            ? (Number(quantity) * Number(rate)).toFixed(2)
+            : (0).toFixed(2);
+    };
+
+    return (
+        <div className="d-flex flex-row w-100 small gap-1">
+            <div className="col-3">
+        <textarea
+            name="textarea"
+            className="w-100"
+            rows="4"
+            placeholder="Optional description"
+            value={itemInfo["description"]}
+            required
+            onChange={(e) => {
+                {
+                    setItemInfo({...itemInfo, description: e.target.value});
+                    let newItemInfo = [...invoiceObj];
+                    newItemInfo[index].description = e.target.value;
+                    setInvoiceObj(newItemInfo);
+                }
+            }}
+        ></textarea>
+            </div>
+            <div className="col-2">
+                <input
+                    type="number"
+                    min="0"
+                    value={invoiceObj[index].quantity}
+                    className="w-100 p-1"
+                    placeholder="Quantity"
+                    required
+                    onChange={(e) => {
+                            setItemInfo({...itemInfo, quantity: e.target.value});
+                            let newItemInfo = [...invoiceObj];
+                            newItemInfo[index].quantity = e.target.value;
+
+                            let total = rowTotal(itemInfo.rate, e.target.value);
+                            newItemInfo[index].total = total;
+                            setInvoiceObj(newItemInfo);
+                    }}
+                />
+            </div>
+            <div className="col-3">
+                <div className="input-group input-medium">
+          <span className="input-group-text h-25 p-1 bg-secondary-subtle">
+            £
+          </span>
+                    <input
+                        type="number"
+                        min="0"
+                        step="0.01"
+                        required
+                        value={invoiceObj[index].rate}
+                        onChange={(e) => {
+                                setItemInfo({...itemInfo, rate: e.target.value});
+                                let newItemInfo = [...invoiceObj];
+                                newItemInfo[index].rate = e.target.value;
+
+                                let total = rowTotal(e.target.value, itemInfo.quantity);
+                                newItemInfo[index].total = total;
+
+                                setInvoiceObj(newItemInfo);
+                        }}
+                        className="h-25 form-control input-lg p-1"
+                        readOnly=""
+                        placeholder="Rate"
+                    />
```

Comment:
Again, maybe this (and other form elements here) could be split up into sub-components.